pkg/types/chain_reader.go

```diff
@@ -16,23 +16,28 @@ const (
 
 type ChainReader interface {
 	// GetLatestValue gets the latest value....
-	// The params argument can be any object which maps a set of generic parameters into chain specific parameters defined in RelayConfig. It must encode as an object via [json.Marshal].
-	// Typically, would be either an anonymous map such as `map[string]any{"baz": 42, "test": true}}`, a struct with `json` tags, or something which implements [json.Marshaler].
+	// The params argument can be any object which maps a set of generic parameters into chain specific parameters defined in RelayConfig.
+	// It must encode as an object via [json.Marshal] and [github.com/fxamacker/cbor/v2.Marshal].
+	// Typically, would be either a struct with field names mapping to arguments, or anonymous map such as `map[string]any{"baz": 42, "test": true}}`
 	//
-	// returnVal must [json.Unmarshal] as an object, and so should be a map, struct, or implement the [json.Unmarshaler] interface.
+	// returnVal must [json.Unmarshal] and and [github.com/fxamacker/cbor/v2.Marshal] as an object.
 	//
 	// Example use:
 	//  type ProductParams struct {
-	// 		Arg int `json:"arg"`
+	// 		Id int `json:"arg"`
```

Comment:
```suggestion
	// 		ID int `json:"arg"`
```

Reply:
If we're changing the name of the golang field from Arg to ID, shouldn't we also change the json field from "arg" to "ID"?